yt_dlp/extractor/crunchyroll.py

```diff
@@ -733,13 +736,111 @@ class CrunchyrollBetaIE(CrunchyrollBaseIE):
+        episode_data = traverse_obj(initial_state, ('content', 'byId', internal_id))
+        if not self._get_cookies(url).get('etp_rt'):
+            video_id = episode_data.get('external_id').split('.')[1]
+            series_id = traverse_obj(episode_data, ('episode_metadata', 'series_slug_title'))
+            return self.url_result(f'https://www.crunchyroll.com/{lang}{series_id}/{display_id}-{video_id}',
+                                   CrunchyrollIE.ie_key(), video_id)
+
+        app_config = self._parse_json(
+            self._search_regex(r'__APP_CONFIG__\s*=\s*({.+?})\s*;', webpage, 'app config'),
+            display_id)
+        client_id = traverse_obj(app_config, ('cxApiParams', 'accountAuthClientId'))
+        api_domain = traverse_obj(app_config, ('cxApiParams', 'apiDomain'))
+        basic_token = str(base64.b64encode(('%s:' % client_id).encode('ascii')), 'ascii')
+        auth_response = self._download_json(
+            api_domain + '/auth/v1/token', display_id,
+            note='Authenticating with cookie',
+            headers={
+                'Authorization': 'Basic ' + basic_token
+            }, data='grant_type=etp_rt_cookie'.encode('ascii'))
+        policy_response = self._download_json(
+            api_domain + '/index/v2', display_id,
+            note='Retrieving signed policy',
+            headers={
+                'Authorization': auth_response.get('token_type') + ' ' + auth_response.get('access_token')
+            })
+        bucket = traverse_obj(policy_response, ('cms', 'bucket'))
+        params = {
+            'Policy': traverse_obj(policy_response, ('cms', 'policy')),
+            'Signature': traverse_obj(policy_response, ('cms', 'signature')),
+            'Key-Pair-Id': traverse_obj(policy_response, ('cms', 'key_pair_id')),
+            'locale': traverse_obj(initial_state, ('localization', 'locale'))
+        }
+        episode_response = self._download_json(
+            api_domain + '/cms/v2' + bucket + '/episodes/' + internal_id, display_id,
+            note='Retrieving episode metadata',
+            query=params)
+        stream_response = self._download_json(
+            episode_response.get('playback'), display_id,
+            note='Retrieving stream info')
+
+        thumbnails = []
+        for thumbnails_data in traverse_obj(episode_response, ('images', 'thumbnail')):
+            for thumbnail_data in thumbnails_data:
+                thumbnails.append({
+                    'url': thumbnail_data.get('source'),
+                    'width': thumbnail_data.get('width'),
+                    'height': thumbnail_data.get('height'),
+                })
+        subtitles = {}
+        for lang, subtitle_data in stream_response.get('subtitles').items():
+            subtitles[lang] = [{
+                'url': subtitle_data.get('url'),
+                'ext': subtitle_data.get('format')
+            }]
+
+        requested_languages = self._configuration_arg('language')
```

Comment:
Yeah, this is a case of copying code without really understanding it. I just ripped that out of VRVIE verbatim.

Is there a cleaner way to keep the hardsubs from getting selected as default? (I doubt very many people ever want the hardsubbed versions, given that softsubs are provided)

Should I just ignore hardsubs generally? Those extra http requests slow down the extractor and are probably almost never needed.

Reply:
oh! I thought u copied it from CrunchyrollIE. But now I see it doesnt have any extractor args defined. Let me look over this code again

Reply:
you are currently not using the extractor args to reduce requests? or am I missing something?

Reply:
> Is there a cleaner way to keep the hardsubs from getting selected as default? (I doubt very many people ever want the hardsubbed versions, given that softsubs are provided)

yes, making it not selected by default is easy. But that won't reduce the http requests

> Should I just ignore hardsubs generally? Those extra http requests slow down the extractor and are probably almost never needed.

I'm sure someone will ask for it. And since the extractor args already exist for CrunchyrollIE, might as well do the same here